catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -0,0 +1,109 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.metacard.validation;
+
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_ERRORS;
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_WARNINGS;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class MetacardValidityFilterPlugin implements PostQueryPlugin {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(MetacardValidityFilterPlugin.class);
+
+    private static Map<String, List<String>> attributeMap = new HashMap<>();
+
+    public static final String INVALID_STATE = "invalid_state";
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Result> results = input.getResults();
+
+        // process security
+        for (Result result : results) {
+            Metacard metacard = result.getMetacard();
+            Attribute securityAttribute = metacard.getAttribute(Metacard.SECURITY);
+            HashMap<String, List<String>> securityMap = null;
+            Serializable tempSecurityMap = null;
+            if (securityAttribute != null) {
+                tempSecurityMap = securityAttribute.getValue();
+                try {
+                    securityMap = (HashMap<String, List<String>>) tempSecurityMap;
+
+                } catch (Exception e) {
+                    LOGGER.warn("Security was not the expected type of Map<String, List<String>>",
+                            e);
+                }
+            } else {
+                securityMap = new HashMap<>();
+            }
+
+            try {
+                if (metacard.getAttribute(VALIDATION_ERRORS).getValues() != null
+                        || metacard.getAttribute(VALIDATION_WARNINGS).getValues() != null) {
+                    for (Map.Entry<String, List<String>> attributeMapping : attributeMap
+                            .entrySet()) {
+                        securityMap.put(attributeMapping.getKey(), attributeMapping.getValue());
+                    }
+
+                }
+            } catch (NullPointerException e) {
+                LOGGER.debug("Allowed NPE when searching for {} and {}", VALIDATION_ERRORS,
+                        VALIDATION_WARNINGS, e);
+            } finally {
+                if (!securityMap.isEmpty()) {
```

Comment:
Yeah normally you should just check for the nulls instead of using exceptions for logic flow. Also if it throws nullpointer above couldn't the finally branch also NPE?